embedded/remotestorage/s3/s3.go

```diff
@@ -71,7 +88,12 @@ func Open(
 	// Bucket must have no '/' at all
 	bucket = strings.Trim(bucket, "/")
 	if strings.Contains(bucket, "/") {
-		return nil, ErrInvalidArguments
+		return nil, ErrInvalidArgumentsBucketSlash
+	}
+
+	// Bucket name must not be empty
+	if bucket == "" {
+		return nil, ErrInvalidArgumentsBucketEmpty
 	}
 
 	// if prefix is not empty, it must end with '/'
```

Comment:
is it safe for prefixes to contain slashes?

Reply:
yes, like if you want to scan sub-folders, e.g. `defaultdb/index/`